assets/javascripts/admin_needle.js

```diff
@@ -1,88 +1,88 @@
 function setupAdminNeedles() {
     function ajaxUrl() {
-	var url = $('#needles').data('ajax-url');
-	return url + "?last_match=" + $('#last_match_filter').val() + "&last_seen=" + $('#last_seen_filter').val();
+        var url = $('#needles').data('ajax-url');
+        return url + "?last_match=" + $('#last_match_filter').val() + "&last_seen=" + $('#last_seen_filter').val();
     }
-    
+
     var table = $('#needles').DataTable(
-	{ "ajax": ajaxUrl(),
-	  deferRender: true,
-	  "columns": [
-	      { "data": "directory" },
-	      { "data": "filename" },
-	      { "data": "last_seen" },
-	      { "data": "last_match" }
-	  ],
-	  
-	  "order": [[0, "asc"], [1, "asc"]] ,
-          "columnDefs": [
-              {  "targets": [2,3],
-                 "className": "time",
-		 "render": function ( data, type, row ) {
-		     if (type === 'display' && data != 'never') {
+        { "ajax": ajaxUrl(),
+            deferRender: true,
+            "columns": [
+            { "data": "directory" },
+            { "data": "filename" },
+            { "data": "last_seen" },
+            { "data": "last_match" }
+            ],
+            "processing": true,
```

Comment:
Here's the actual change - this will enable the progress indication provided by DataTable.